librz/asm/arch/8051/8051_disas.c

```diff
@@ -95,13 +95,13 @@ static char *rz_8051_disas(ut64 pc, const ut8 *buf, int len, int *olen) {
 		case 2:
 			if (len > 1) {
 				if (arg1 == A_OFFSET) {
-					disasm = rz_str_newf(name, arg_offset(pc + 2, buf[1]));
+					disasm = rz_str_newf(name, (unsigned int)arg_offset(pc, pc + 2, buf[1]));
```

Comment:
What about using shorter types instead?

Reply:
What do you mean by shorter types? We explicitly want to support more than 16bit here. This will still chop >32bit but I would like to avoid touching the format strings in there before reviewing and testing the entire architecture implementation.

Reply:
I meant `utXX`, since `strlen("ut16")` < `strlen("unsigned int")`. Impossibly pedantic nitpick, I know.

Reply:
Ah I see. No, we can't use `ut32` since the format strings use `%x` which is defined as (unsigned) `int`.

Reply:
Ok, no worries then.